.github/workflows/tests.yaml

```diff
@@ -0,0 +1,64 @@
+name: Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches:
+      - "*"
+
+  schedule:
+    - cron: "59 23 * * 3"
+
+jobs:
+  Linting:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v4
+      - uses: pre-commit/action@v3.0.0
+
+  Tests:
+    name: ${{ matrix.os }}, ${{ matrix.environment-file }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        environment-file: [ci/latest.yaml]
+        include:
+          - environment-file: ci/dev.yaml
+            os: ubuntu-latest
+
+    steps:
+      - name: checkout repo
+        uses: actions/checkout@v2
+
+      - name: setup micromamba
+        uses: mamba-org/provision-with-micromamba@main
+        with:
+          environment-file: ${{ matrix.environment-file }}
+          micromamba-version: "latest"
+          channel-priority: "flexible"
+
+      - name: Install xvec
+        shell: bash -l {0}
+        run: pip install .
+        if: matrix.os != 'windows-latest'
+
+      - name: Install xvec
+        shell: powershell
+        run: pip install .
+        if: matrix.os == 'windows-latest'
```

Comment:
`windows-latest` also has `bash` pre-installed, so you should be able to remove this?